index.ts

```diff
@@ -90,7 +101,10 @@ async function run() {
 try {
   await run();
 } catch (err) {
-  if (err instanceof UserError) {
+  if (err instanceof Deno.errors.PermissionDenied) {
```

Comment:
👏 Good call on these. Just thinking out loud (no need to change anything here), it would be nice if the shelling-out-to-git piece failed gracefully, if someone didn't want to use that feature and/or provide that permission. Improving the error handling is good but doesn't fix the fact that you would end up in a weird state if it failed here.

Reply:
I thought about that too, and just encountered it.

![Screen Shot 2021-06-28 at 10 05 47 AM](https://user-images.githubusercontent.com/10974/123659580-77668200-d7f8-11eb-8852-4b6b0770a2db.png)


It may work to just create a GitError like your UserError which then just prints out just the message. In this case where I'm calling `init` on an uninitialized git repo the error message was probably enough to resolve it.